File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        n = len(points)
        dist = []
        for i in range(n):
            x,y = points[i]
            dist.append([x*x+y*y,x,y])
        
        heapq.heapify(dist)
        res = []
        while k>0:
            d,x,y = heapq.heappop(dist)
            res.append([x,y])
            k-=1
        return res

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        int n = points.length;
        int[][] res = new int[k][2];
        
        PriorityQueue<int[]> q = new PriorityQueue<>((int[] p1, int[] p2)->Integer.compare(p1[0],p2[0]));
        
        for(int i=0;i<n;i++){
            int x = points[i][0], y=points[i][1];
            int[] pair = new int[]{x*x+y*y,i};
            q.add(pair);
        }

        for(int i=0;i<k;i++){
            int[] p = q.poll();
            res[i] = points[p[1]];
        }
        return res;
    }
}